ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/Snapshots.hs

```diff
@@ -116,14 +118,14 @@ takeSnapshot tracer hasFS encLedger oldest =
         return Nothing
       NotOrigin tip -> do
         let number   = unSlotNo (realPointSlot tip)
-            snapshot = DiskSnapshot number Nothing
+            snapshot = DiskSnapshot number Nothing Nothing
         snapshots <- listSnapshots hasFS
         if List.any ((== number) . dsNumber) snapshots then
           return Nothing
         else do
-          encloseTimedWith (TookSnapshot snapshot tip >$< tracer)
-              $ writeSnapshot hasFS encLedger snapshot oldest
-          return $ Just (snapshot, tip)
+          snapshotWithCRC <- writeSnapshot hasFS encLedger snapshot oldest
+          encloseTimedWith (TookSnapshot snapshotWithCRC tip >$< tracer) $ pure ()
```

Comment:
This `encloseTimedWith` no longer encloses the time it takes to write the snapshot.

Reply:
A variant of `encloseTimedWith` that allows to return a result seems sensible, as mentioned in https://github.com/IntersectMBO/ouroboros-consensus/pull/1183#discussion_r1679251466